Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/DbCommandBuilders/QueryDbCommandBuilderTest.cs

```diff
@@ -66,21 +131,100 @@ public void Create ()
       var dbCommandFactoryStub = new Mock<IDbCommandFactory>();
       dbCommandFactoryStub.Setup(stub => stub.CreateDbCommand()).Returns(dbCommandStub.Object);
 
+      var commandBuilder = new QueryDbCommandBuilder("Statement #param1 #param2", [ _textQueryParameterWithDataDefinition ], GetSqlDialectStub());
+      var result = commandBuilder.Create(dbCommandFactoryStub.Object);
+
+      dataParameterCollectionStrictMock.Verify();
+      _valueParameterDefinition.Verify();
+      _textParameterDefinition.Verify();
+      Assert.That(result.CommandText, Is.EqualTo("Statement #param1 test"));
+    }
+
+    [Test]
+    public void Create_AddsValueParameter ()
+    {
+      var dataParameterCollectionStrictMock = new Mock<IDataParameterCollection>(MockBehavior.Strict);
+
+      var dbCommandStub = new Mock<IDbCommand>();
+      dbCommandStub.Setup(stub => stub.Parameters).Returns(dataParameterCollectionStrictMock.Object);
+      dbCommandStub.SetupProperty(stub => stub.CommandText);
+
+      var dbCommandFactoryStub = new Mock<IDbCommandFactory>();
+      dbCommandFactoryStub.Setup(stub => stub.CreateDbCommand()).Returns(dbCommandStub.Object);
+
       var dbDataParameterStub = new Mock<IDbDataParameter>();
-      _valueParameterDefinition
-          .Setup(mock => mock.CreateDataParameter(dbCommandStub.Object, _valueQueryParameter.Name, 5))
+
+      _valueParameterDefinition.Setup(mock => mock.GetParameterValue(5)).Returns(5);
+      _valueParameterDefinition.Setup(mock => mock.CreateDataParameter(dbCommandStub.Object, _valueQueryParameter.Name, _valueQueryParameter.Value))
```

Comment:
```suggestion
      _valueParameterDefinition.Setup(mock => mock.CreateDataParameter(dbCommandStub.Object, _valueQueryParameter.Name, 5))
```
Keeping this as `5` makes it easier to reason over.